tools/manifest/vcs.py

```diff
@@ -1,22 +1,39 @@
+import abc
 import json
 import os
 import stat
 from collections import deque
+from collections import MutableMapping
+
+from six import with_metaclass, PY2
 
 from .sourcefile import SourceFile
 from .utils import git
 
+try:
+    from ..gitignore import gitignore
+except ValueError:
+    # relative import beyond toplevel throws *ValueError*!
+    from gitignore import gitignore  # type: ignore
```

Comment:
Can we avoid this at all?

Reply:
We can replicate all of our `sys.path` manipulation with `mypy_path` in our configuration, but that's also painful. Really we should just stop doing crazy stuff with `sys.path`.